packages/engine/Source/Shaders/Model/ModelFS.glsl

```diff
@@ -45,6 +45,10 @@ void main()
     MetadataStatistics metadataStatistics;
     metadataStage(metadata, metadataClass, metadataStatistics, attributes);
 
+    //========================================================================
+    // When not picking metadata START
+    #ifndef METADATA_PICKING_ENABLED
```

Comment:
Is it safe to skip `materialStage`, `customShaderStage`, and `modelColorStage`?
I'm thinking about the case where the custom shader hides some parts of the model, or makes them transparent. If we skip the custom shader, those parts will now be "visible" to metadata picking.
In this case, the user might think they are selecting some visible object in the background, but the metadata that is picked would actually be coming from an invisible object in the foreground.

Reply:
That's a valid point, and I don't have a quick answer, and it may warrant a few more considerations (so I'll address that in a non-inlined comment below the PR)